controllers/handler/nodenetworkconfigurationpolicy_controller.go

```diff
@@ -157,14 +149,39 @@ func (r *NodeNetworkConfigurationPolicyReconciler) Reconcile(_ context.Context,
 		err = r.deleteEnactmentForPolicy(request.NamespacedName.Name)
 		return ctrl.Result{}, err
 	}
+	return r.reconcileEnactmentConditions(instance, &log)
+}
+
+func (r *NodeNetworkConfigurationPolicyReconciler) fetchNodeNetworkConfigurationPolicyInstance(
+	instance *nmstatev1.NodeNetworkConfigurationPolicy,
+	request *ctrl.Request,
+	log *logr.Logger,
+) error {
+	err := r.Client.Get(context.TODO(), request.NamespacedName, instance)
+	if err != nil {
+		if apierrors.IsNotFound(err) {
+			log.Info("Policy is not found, removing previous enactment if any")
+			err = r.deleteEnactmentForPolicy(request.NamespacedName.Name)
+			return err
+		}
+		log.Error(err, "Error retrieving policy")
+		// Error reading the object - requeue the request.
+		return err
+	}
+	return nil
+}
 
+func (r *NodeNetworkConfigurationPolicyReconciler) reconcileEnactmentConditions(
```

Comment:
IMO, it doesn't make sense to call this reconcile*
as in k8s , reconcile is a method of a reconciler, that is called by a controller.